packages/esm-appointments-app/src/form/appointments-form.resource.ts

```diff
@@ -0,0 +1,126 @@
+import dayjs from 'dayjs';
+import useSWR from 'swr';
+import { openmrsFetch } from '@openmrs/esm-framework';
+import {
+  type AppointmentPayload,
+  type AppointmentService,
+  type AppointmentsFetchResponse,
+  type RecurringAppointmentsPayload,
+} from '../types';
+import isToday from 'dayjs/plugin/isToday';
+dayjs.extend(isToday);
+
+const appointmentsSearchUrl = `/ws/rest/v1/appointments/search`;
+
+export function useAppointments(patientUuid: string, startDate: string, abortController: AbortController) {
+  /*
+    SWR isn't meant to make POST requests for data fetching. This is a consequence of the API only exposing this resource via POST.
+    This works but likely isn't recommended.
+  */
+  const fetcher = () =>
+    openmrsFetch(appointmentsSearchUrl, {
+      method: 'POST',
+      signal: abortController.signal,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: {
+        patientUuid: patientUuid,
+        startDate: startDate,
+      },
+    });
+
+  const { data, error, isLoading, isValidating, mutate } = useSWR<AppointmentsFetchResponse, Error>(
+    appointmentsSearchUrl,
+    fetcher,
+  );
+
+  const appointments = data?.data?.length ? data.data : null;
+
+  const pastAppointments = appointments
+    ?.sort((a, b) => (b.startDateTime > a.startDateTime ? 1 : -1))
```

Comment:
This is different from the others. is the descending sort intentional?

Reply:
I just copied this code over from the patient chart, but will test and review. I could see it making sense that you'd want descending on some views and ascending in others, maybe I need to look closer into that.  For instance, in a simple list of a patient's past appointments you might want the most recent in the past first, but for any daily view you probably want the earliest first.